fedimint-client/src/transaction/sm.rs

```diff
@@ -107,19 +108,27 @@ impl State for TxSubmissionStates {
 impl TxSubmissionStates {
     async fn trigger_created_rejected(tx: Transaction, context: DynGlobalClientContext) -> String {
         loop {
-            match context.api().submit_transaction(tx.clone()).await {
-                Ok(serde_result) => match serde_result.try_into_inner(context.decoders()) {
-                    Ok(result) => {
-                        if let Err(transaction_error) = result {
-                            return transaction_error.to_string();
-                        }
+            if SUBMIT_ENDPOINT_API_VERSION <= context.core_api_version() {
+                if let Err(e) = context.api().submit_transaction(tx.clone()).await {
+                    if let Some(call_error) = e.try_to_call_error(context.num_peers()) {
```

Comment:
Switching to `try_to_call_error` changes the semantics here and relaxes consensus requirements. Previously peers had to agree on the error data, now it can be only the error code in the worst case. Do we want that? Have we thought of all the potential side-effects?